### ISSUE TITLE
Test Coverage Fails Due to Broken @cover Statements

### ISSUE THREAD

**Issue by gordonbanderson** [silverstripe/silverstripe-framework#4904]
There are a small number of unit tests failing when running PHPUnit whilst simultaneously creating coverage reports for upload to third party coverage services.  These are

```bash
There were 2 errors:
1) GridFieldTest::testGridFieldDefaultConfig
PHP_CodeCoverage_Exception: Trying to @cover not existing method "GridField::setComponents".
/home/travis/builds/ss/vendor/phpunit/phpunit/composer/bin/phpunit:63
2) GridFieldTest::testGridFieldSetCustomConfig
PHP_CodeCoverage_Exception: Trying to @cover not existing method "GridField::setComponents".
/home/travis/builds/ss/vendor/phpunit/phpunit/composer/bin/phpunit:63

1) StringFieldTest::testForTemplate
PHP_CodeCoverage_Exception: Trying to @cover not existing class or interface "StringField->forTemplate".
/home/travis/builds/ss/vendor/phpunit/phpunit/composer/bin/phpunit:63
2) StringFieldTest::testLowerCase
PHP_CodeCoverage_Exception: Trying to @cover not existing class or interface "StringField->LowerCase".
/home/travis/builds/ss/vendor/phpunit/phpunit/composer/bin/phpunit:63
3) StringFieldTest::testUpperCase
PHP_CodeCoverage_Exception: Trying to @cover not existing class or interface "StringField->UpperCase".
```

To recreate clone https://github.com/gordonbanderson/silverstripe-framework/tree/TESTEVAL into another repository that is configured to build from Travis and then push to the repository to instigate a build.

**Comment by gordonbanderson** [comment]
The code coverage report can be viewed here, it includes coverage uploaded when failure occurs.  It does make for interesting reading https://codecov.io/github/gordonbanderson/silverstripe-framework?ref=f78d75159a3060a3b98e102982f262ef09156044

**Comment by madmatt** [comment]
Fixed with #5229 and #5240 :)
